### PR TITLE
Instrument daemon with metrics

### PR DESCRIPTION
**Pull request by Crevil**
This change adds prometheus metrics to the daemon implementation giving insights
into how they run.

### CODE REVIEW

File: internal/daemon/daemon_test.go [hunk 1]
```diff
@@ -7,20 +7,31 @@ import (
...
 	testDuration := 100 * time.Millisecond
 	expectedTickCount := int32(10)
 
 	var actualTickCount int32
 	d := daemon.New(daemon.Configuration{
 		Interval: tickInterval,
-		Logger:   logger,
+		Probes: &daemon.Probes{
+			Started: func(time.Duration) {
+				t.Log("Daemon started")
```
Comment (emilingerslev) [inline internal/daemon/daemon_test.go]: Would be nice with the `started`, `stopped`, etc. as a field in the log.

Comment (Crevil) [reply internal/daemon/daemon_test.go]: In a test?

**Comment by Crevil** [comment]
You're right. This really is just a reporter. I'll rename it.

File: internal/daemon/daemon_test.go (hunk 1 above)
Comment (Crevil) [reply internal/daemon/daemon_test.go]: I've added a `state` field to the main logs instead with values `started` and `stopped`.

**Pull request merged**